src/ViewRenderer.php

```diff
@@ -561,7 +561,7 @@ private function extractControllerName(object $controller): string
             return $cache[$class];
         }
 
-        $regexp = '/((?<=controller\\\|s\\\)(?:[\w\\\]+)|(?:[a-z]+))controller/iuU';
+        $regexp = '/((?<=controller\\\|controllers\\\)(?:[\w\\\]+)|(?:[a-z]+))controller/iuU';
```

Comment:
```suggestion
        $regexp = '/((?<=controllers?\\\)(?:[\w\\\]+)|(?:[a-z]+))controller/iuU';
```

Reply:
It's not work: `preg_match(): Compilation failed: lookbehind assertion is not fixed length at offset 1`

Reply:
I suggests:
https://regex101.com/r/5M2D54/3
```php
$regexp = '!(?:controllers?\\(?<subpath>[^\\]+(?:\\[^\\]+)*\\))?(?<path>(?:[^\\]+))controller$!iu';
if(preg_match($regexp, $class, $m) !== 1) {
  throw ...
}
$name=($m['subpath'] ?? '') . $m['path'];
```

Reply:
Unfortunately the last snippet doesn't work too. Gives the following error:

```
preg_match(): Compilation failed: missing terminating ] for character class at offset 76
```

Also it's more complicated and harder to read / maintain in my opinion. With the amount of variants and their length ("controller" and "controller**s**"), duplicating the whole world with different ending is acceptable and a better option I think.